### ISSUE TITLE
Petitboot timeout function would be no function while the network cable plug in

### ISSUE THREAD

**Issue by deathscytheh28** [open-power/petitboot#14]
We found the timeout function in Petitboot would not count down if we plug in the network cable when system power on, and the auto-boot function seems no function, too. (ps. our system use PXE boot, no hard drive)

**Comment by sammj** [comment]
Please post the Petitboot logs under /var/log/petitboot/ - perhaps something is cancelling the autoboot countdown.

**Comment by deathscytheh28** [comment]
The attachment is the pb-discover.log and pb-udhcpc.log
[pb-discover.txt](https://github.com/open-power/petitboot/files/189192/pb-discover.txt)
[pb-udhcpc.txt](https://github.com/open-power/petitboot/files/189194/pb-udhcpc.txt)

**Comment by jk-ozlabs** [comment]
```
Running command:
 exe:  /usr/bin/tftp
 argv: '/usr/bin/tftp' '-g' '-l' '/tmp/pb-fa9Tzk' '-r' '/pxelinux.cfg/01-00-01-6c-80-03-c2' '10.36.239.30' '69'
Sending discover...
Sending discover...
Sending discover...
tftp: timeout
```

It looks like your DHCP server's response is telling petitboot that there are boot options available on the network (via TFTP), but connections to that TFTP server are timing out. This means that petitboot will be waiting for that TFTP to complete.

This DHCP response is coming in on the interface with MAC address 00:01:6c:80:03:c2

To fix this, you'll need to configure your DHCP server correctly.

**Comment by deathscytheh28** [comment]
We did another experiment that we install a hard drive and also plug in network cable. From the Petitboot log ([pb-discover_30.txt](https://github.com/open-power/petitboot/files/193232/pb-discover_30.txt)), we saw the hard drive was be set as default boot option and timeout 10 sec, and the TFTP timeout message also appeared. 

In this situation, we wait almost 30 sec but still not boot into hard drive, does this behavior right?
